### PR TITLE
Scripts refactoring

### PR DESCRIPTION
**Pull request by aj-maz**
wip

### CODE REVIEW

**Comment by aj-maz** [comment]
I have changed the architecture of our scripts. 

Each script contains a single action

Actions are composable functions that are building blocks of our scripts. Composed actions are also considered an action so they can be also composed. 

This way we gonna have a lot more maintainable and flexibility in writing scripts.

I have written a utility pipe function that will compose actions.

a typical scripts pipe looks like this:


```

await pipeActions({ mintTo, xCitadelFees, user, multisig })(
    setupAndDeploy,
    () => console.log("Contracts setted up ..."),
    mintForknet,
    () => console.log("Forknet mints ..."),
    getRoleSigners,
    () => console.log("Roles assigned ..."),
    initializer,
    () => console.log("Contracts initialized ..."),
    setXCitadelStrategy,
    () => console.log("Setted xCitadel strategy  ..."),
    grantRoles,
    () => console.log("Roles Granted ..."),
    citadelMinterSetup,
    () => console.log("Citadel minter setup ..."),
    approveFundingTokens,
    () => console.log("Funding tokens approved ..."),
    medianOracleUpdatePrice,
    () => console.log("Median oracle update the price ..."),
    setDiscount,
    () => console.log("Discount setted ..."),
    bondTokenForXCTDL,
    () => console.log("bond some WBTC and CVX to get xCTDL ..."),
    xCTDLVesting,
    () => console.log("xCTDL vesting"),
    setupKnightingRound,
    () => console.log("Knighting round setup")
  );

```

**Comment by aj-maz** [comment]
will resolve #147 and #104

File: scripts/actions/bondTokenForXCTDL.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+const hre = require("hardhat");
+const ethers = hre.ethers;
+const { address } = require("../utils/helpers");
+
+const { formatUnits, parseUnits } = ethers.utils;
+
+const bondTokenForXCTDL = async ({
```
Comment (dapp-whisperer) [inline scripts/actions/bondTokenForXCTDL.js:7]: imo this should be a generalized macro - it should bond for one token (any given token)

**Comment by mrbasado** [comment]
Add KNIGHTING_ROUND_BVECVX

**Review by dapp-whisperer — APPROVED** [review]

**Comment by dapp-whisperer** [comment]
I believe the tests are failing due to known fork issue, confirmed with basado to merge.

Will follow up with comments or issues in future PR

**Pull request merged**